docs/devices_api.yml

```
@@ -26,4 +26,140 @@ security:
 tags:
   - name: DevicesAPI
 
-paths: {} # TBD
+paths:
+  /configuration:
+    get:
+      operationId: Get Device Configuration
+      tags:
+        - DevicesAPI
+      summary: Query the configuration store; retrieve all key-value pairs
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  id:
+                    type: string
+                    format: uuid
+                  configured:
+                    type: object
+                    $ref: '#/components/schemas/Configuration'
+                  reported:
+                    type: object
+                    $ref: '#/components/schemas/Configuration'
+                  reported_ts:
+                    type: string
+                    format: date-time
+                  updated_ts:
+                    type: string
+                    format: date-time
```

I guess the only informative part from the device's point of view is the configured attributes as it can resolve the other attributes (with the exception of `updated_ts`)
```suggestion
                  type: object
                  $ref: '#/components/schemas/Configuration'
```

---

@alfrunes are you suggesting returning the configuration only, or keep configured & updated_ts only?

---

Only the `configuration`, I don't think the `updated_ts` will be of much use for the device.

---

Done!